Formula/openttd.rb

```diff
@@ -18,11 +18,24 @@ class Openttd < Formula
   end
 
   depends_on "cmake" => :build
+  depends_on "ninja" => :build
   depends_on "libpng"
   depends_on "lzo"
   depends_on macos: :high_sierra # needs C++17
   depends_on "xz"
 
+  on_linux do
+    depends_on "fluid-synth"
+    depends_on "fontconfig"
+    depends_on "freetype"
+    depends_on "gcc"
+    depends_on "mesa"
+    depends_on "mesa-glu"
+    depends_on "sdl2"
```

Comment:
I would recommend just sticking to Unix Makefiles unless there is a particular reason for `ninja`. That is our usually preference.

Also, if you don't remove `fixup_bundle`, then dependency become build-only on macOS, i.e. the dylibs are copied into the App, since the result is a portable App:
```console
% otool -L  /usr/local/opt/openttd/OpenTTD.app/Contents/MacOS/openttd
/usr/local/opt/openttd/OpenTTD.app/Contents/MacOS/openttd:
	@executable_path/../Frameworks/libpng16.16.dylib (compatibility version 54.0.0, current version 54.0.0)
	/usr/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.11)
	@executable_path/../Frameworks/liblzma.5.dylib (compatibility version 8.0.0, current version 8.5.0)
	@executable_path/../Frameworks/liblzo2.2.dylib (compatibility version 3.0.0, current version 3.0.0)
	/System/Library/Frameworks/OpenGL.framework/Versions/A/OpenGL (compatibility version 1.0.0, current version 1.0.0)
        ...

% otool -L  /usr/local/opt/openttd/OpenTTD.app/Contents/Frameworks/liblzma.5.dylib
/usr/local/opt/openttd/OpenTTD.app/Contents/Frameworks/liblzma.5.dylib:
	/usr/local/opt/openttd/OpenTTD.app/Contents/Frameworks/liblzma.5.dylib (compatibility version 8.0.0, current version 8.5.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.0.0)

% objdump --rpaths --macho /usr/local/opt/openttd/OpenTTD.app/Contents/Frameworks/liblzma.5.dylib
/usr/local/opt/openttd/OpenTTD.app/Contents/Frameworks/liblzma.5.dylib:
```

In other formulae, we have preferred breaking `fixup_bundle` (e.g. manually relinking in `ppsspp` and disabling fixup in `minetest`). If you keep `fixup_bundle` then the dependencies should probably account for that e.g.
```suggestion
  depends_on "libpng" => :build
  depends_on "lzo" => :build
  depends_on macos: :high_sierra # needs C++17
  depends_on "xz" => :build

  on_linux do
    depends_on "fluid-synth"
    depends_on "fontconfig"
    depends_on "freetype"
    depends_on "gcc"
    depends_on "libpng"
    depends_on "lzo"
    depends_on "mesa"
    depends_on "mesa-glu"
    depends_on "sdl2"
    depends_on "xz"
```